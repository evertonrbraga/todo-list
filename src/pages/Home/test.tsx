import { render, screen } from 'project-testing-library'
import { Home } from '.'

describe('<Home />', () => {
  it('should snapshot the Home page', () => {
    const { container } = render(<Home />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the todo logo correctly', () => {
    render(<Home />)
    const svgImg = 'test-file-stub'
    const altText = 'Logo ToDo'
    const imageElement = screen.getByAltText(altText)
    expect(imageElement).toHaveAttribute('src', svgImg)
  })

  it('should render the input correctly', () => {
    render(<Home />)
    const placeholder = 'Adicione uma nova tarefa'
    const input = screen.getByPlaceholderText(placeholder)
    expect(input).toBeInTheDocument()
  })

  it('should check if the created tasks control works properly', () => {
    render(<Home />)
    const taskLabel = screen.getByText('Tarefas criadas')
    const taskQuantity = screen.getAllByText('0')[0]
    expect(taskLabel).toBeInTheDocument()
    expect(taskQuantity).toHaveTextContent('0')
  })

  it('should check if the finished tasks control works properly', () => {
    render(<Home />)
    const finishedLabel = screen.getByText('Conclu√≠das')
    const finishedQuantity = screen.getAllByText('0')[1]
    expect(finishedLabel).toBeInTheDocument()
    expect(finishedQuantity).toHaveTextContent('0')
  })
})
