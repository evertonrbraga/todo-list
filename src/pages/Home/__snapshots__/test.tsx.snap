// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should snapshot the Home page 1`] = `
.c0 {
  margin: 0;
  height: 200px;
  width: 100%;
  position: absolute;
  background-color: #0D0D0D;
}

<div
  class="c0"
/>
`;
